org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/core/IDisconnectable.java

```diff
@@ -20,6 +20,7 @@
  * @author Matthew Gerring
  *
  */
+@FunctionalInterface

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Annotate the "IDisconnectable" interface with the @FunctionalInterface annotation (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1609)